Sources/Classes/Domain/Models/RSConfig.swift

```diff
@@ -70,23 +77,23 @@ open class RSConfig: NSObject {
         return _gzipEnabled
     }
     
-    @objc
-    public init(writeKey: String) {
-        _writeKey = writeKey
+    private var _dataResidencyServer: RSDataResidencyServer = .US
+    public var dataResidencyServer: RSDataResidencyServer {
+        return _dataResidencyServer
     }
     
-    @discardableResult @objc
-    public func dataPlaneURL(_ dataPlaneUrl: String) -> RSConfig {
-        guard let url = URL(string: dataPlaneUrl), let scheme = url.scheme, let host = url.host else {
-            Logger.logError("dataPlaneUrl is invalid")
-            return self
-        }
-        if let port = url.port {
-            _dataPlaneUrl = "\(scheme)://\(host):\(port)"
+    @objc
+    public init(writeKey: String, dataPlaneURL: String) {
+        _writeKey = writeKey
+        if let url = URL(string: dataPlaneURL), let scheme = url.scheme, let host = url.host {
+            if let port = url.port {
+                _dataPlaneURL = "\(scheme)://\(host):\(port)"
```

Comment:
Shouldn't we also use path of the URL ? (Similar to the control plane URL Usecase.)

Reply:
This is the legacy way. I'm not sure why we used it like that. Can you tell me the reason behind that?

Reply:
I do not think we have a reason why we ignored the path previously, but if we ignore path this may create issue for the custom data plane urls passed by the users. Could be possible if some one uses OSS deployment.

Reply:
Resolved.